versions.lock

```diff
@@ -0,0 +1,253 @@
+com.sun.jersey.contribs:jersey-guice:1.9 (4 constraints: 65529ed7)
+com.sun.xml.bind:jaxb-impl:2.2.3-1 (1 constraints: 330c2404)
+com.thoughtworks.paranamer:paranamer:2.8 (3 constraints: aa296195)
+com.twitter:chill-java:0.9.3 (2 constraints: a716716f)
+com.twitter:chill_2.11:0.9.3 (2 constraints: 121b92c3)
+com.twitter:parquet-hadoop-bundle:1.6.0 (3 constraints: 7c262424)
+com.univocity:univocity-parsers:2.7.3 (1 constraints: c40ccb27)
+com.vlkan:flatbuffers:1.2.0-3f79e055 (2 constraints: 411e1dee)
+commons-beanutils:commons-beanutils:1.7.0 (1 constraints: da0e635f)
+commons-beanutils:commons-beanutils-core:1.8.0 (1 constraints: 1d134124)
+commons-cli:commons-cli:1.2 (8 constraints: 9467c282)
+commons-codec:commons-codec:1.10 (18 constraints: cfeebaf2)
+commons-collections:commons-collections:3.2.2 (4 constraints: 42476934)
+commons-configuration:commons-configuration:1.6 (1 constraints: 2d0d5c14)
+commons-daemon:commons-daemon:1.0.13 (1 constraints: d50c811c)
+commons-dbcp:commons-dbcp:1.4 (3 constraints: 9029e0e4)
+commons-digester:commons-digester:1.8 (1 constraints: bf1228fe)
+commons-el:commons-el:1.0 (1 constraints: fb077074)
+commons-httpclient:commons-httpclient:3.1 (4 constraints: e52cc77f)
+commons-io:commons-io:2.4 (6 constraints: 4a568049)
+commons-lang:commons-lang:2.6 (20 constraints: c11a7ff4)
+commons-logging:commons-logging:1.2 (27 constraints: f2ab390b)
+commons-net:commons-net:3.1 (3 constraints: 3d222e61)
+commons-pool:commons-pool:1.6 (4 constraints: e336ab5e)
+dk.brics.automaton:automaton:1.11-8 (1 constraints: 92088a8d)
+hsqldb:hsqldb:1.8.0.10 (1 constraints: f008499f)
+io.airlift:aircompressor:0.10 (1 constraints: 090a9fb2)
+io.dropwizard.metrics:metrics-core:3.1.5 (6 constraints: 865ea0ba)
+io.dropwizard.metrics:metrics-graphite:3.1.5 (1 constraints: 1a0dc936)
+io.dropwizard.metrics:metrics-json:3.1.5 (1 constraints: 1a0dc936)
+io.dropwizard.metrics:metrics-jvm:3.1.5 (1 constraints: 1a0dc936)
+io.netty:netty:3.9.9.Final (9 constraints: 9eb0396d)
+io.netty:netty-all:4.1.17.Final (3 constraints: d2312526)
+it.unimi.dsi:fastutil:7.0.13 (1 constraints: fc0d4043)
+javax.activation:activation:1.1.1 (1 constraints: 140dbb36)
+javax.annotation:javax.annotation-api:1.2 (2 constraints: 2d21193d)
+javax.inject:javax.inject:1 (4 constraints: 852d0c1a)
+javax.jdo:jdo-api:3.0.1 (2 constraints: 4c1dcc1a)
+javax.servlet:javax.servlet-api:3.1.0 (1 constraints: 150dc436)
+javax.servlet:servlet-api:2.5 (7 constraints: c87e5aa1)
+javax.servlet.jsp:jsp-api:2.1 (1 constraints: 290d5a14)
+javax.transaction:jta:1.1 (1 constraints: 9f07d96b)
+javax.validation:validation-api:1.1.0.Final (1 constraints: 13133130)
+javax.ws.rs:javax.ws.rs-api:2.0.1 (5 constraints: 6e649355)
+javax.xml.bind:jaxb-api:2.2.11 (6 constraints: a069fd48)
+javolution:javolution:5.5.1 (1 constraints: f00d1a43)
+jline:jline:2.12 (3 constraints: 98208776)
+joda-time:joda-time:2.9.9 (5 constraints: c2326fe6)
+junit:junit:4.12 (2 constraints: 3f0d2639)
```

Comment:
I was surprised that junit ends up in the production dependencies, rather than in the `[Test dependencies]` block, but after some digging it turns out `pig` depends on it:
```
$ ./gradlew why --hash 3f0d2639
...
> Task :why
junit:junit:4.12
        projects -> 4.12
        org.apache.pig:pig -> 4.8.1
```

Reply:
That really doesn't seem right - we should probably exclude junit as a transitive from pig.

Reply:
Are you sure pig doesn't require junit at runtime though?

Reply:
I'm not familiar enough with Pig to know for sure - perhaps would like to get feedback on the matter from others more familiar with how Iceberg's interaction with Pig works.

Reply:
Yep, I would be careful removing it even if doesn't make sense, unless you're sure it's not used in any production code path.

Reply:
Pig is a `compileOnly` dependency and it is expected to be provided by the platform where Iceberg is being used, so this should be Ok